index.d.ts

```diff
@@ -18,4 +18,23 @@ for await (const chunk of stream) {
 }
 ```
 */
-export default function mergeStreams(streams: NodeJS.ReadableStream[]): NodeJS.ReadableStream;
+export default function mergeStreams(streams: NodeJS.ReadableStream[]): MergedStreams;
+
+/**
+Single stream combining the output of multiple streams.
+*/
+export type MergedStreams = NodeJS.ReadableStream & {
+	/**
+	Pipe a new readable stream.
+
+	This fails if `MergedStream` has already ended.
+	*/
+	add(stream: NodeJS.ReadableStream): void;
+
+	/**
+	Unpipe a previously added stream.
```

Comment:
Unclear if `remove` also works with streams added in the initializer and not with `add()`.